### PR TITLE
Make local dependency version of orml_nft

### PR DESCRIPTION
**Pull request by justinfrevert**
This copies the `orml_nft` source code and brings it into a local pallet, `anmol-nft`. The goal of this is to bring the NFT logic to a place where we can modify it and add logic for e.g. fractionalization.

All of the code in `pallets/anmol-nft` is unchanged from the `0.4.0` version of `orml-nft`

I still had some questions on this, which I will put below, but I think that this PR at least fulfills what was discussed [here](https://dotmatrix-workspace.slack.com/archives/C022F94ATM2/p1624342716035800?thread_ts=1624316072.033900&cid=C022F94ATM2). A direct follow-on task is to add the logic described [here](https://www.notion.so/substrate-fractional-NFT-783d803a392b4d19a2d3f12b33fa77a3) in `2.`

### CODE REVIEW

File: pallets/anmol-nft/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+[package]
+name = "anmol-nft"
```
Comment (justinfrevert) [inline pallets/anmol-nft/Cargo.toml]: I thought this name was arbitrary enough to start with.

File: pallets/anmol-nft/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+description = "NFT pallet for extended functionality"
+repository = "https://github.com/open-web3-stack/open-runtime-module-library/tree/master/nft"
+license = "Apache-2.0"
+version = "0.1.0"
```
Comment (justinfrevert) [inline pallets/anmol-nft/Cargo.toml]: The original pallet was at `0.4.0`, I didn't think that this made sense to keep and so decided to start it off at a lower version.

**Comment by justinfrevert** [comment]
I had questions on quality process:

1. Are we running `cargo fmt` on the code with this [config](rustfmt)? I noticed a lot of diffs when I did this, so I thought that either we're not doing this or some other code linting process is followed(or this is intended and we just need to run the command). General guidance here is appreciated.
2. Is there a good way to validate the changes? In this case, I expect a no-op since the code isn't actually changing. But in most cases, I would like a way to see and test changes manually.

File: pallets/anmol-nft/Cargo.toml (hunk 1 above)
Comment (justinfrevert) [reply pallets/anmol-nft/Cargo.toml]: I suppose one idea is to name it something more descriptive if we want to share the pallet further down the line. `fractionalizable-nft` or `extended-nft` or something.

Comment (korzewski) [reply pallets/anmol-nft/Cargo.toml]: `anmol-nft` sounds good to me, @nblogist any thoughts?

**Comment by korzewski** [comment]
> I had questions on quality process:
> 
> 1. Are we running `cargo fmt` on the code with this [config](rustfmt)? I noticed a lot of diffs when I did this, so I thought that either we're not doing this or some other code linting process is followed(or this is intended and we just need to run the command). General guidance here is appreciated.
> 2. Is there a good way to validate the changes? In this case, I expect a no-op since the code isn't actually changing. But in most cases, I would like a way to see and test changes manually.

1. We're not using `cargo fmt` yet. Haven't trying it out before, if you see some benefits of using it feel free to create a new ticket with concise description here https://www.notion.so/1b047c821544433d8703f352d347b207?v=7a4133f9bd00425d8f6f7292ad2fe3a3. Is it something like https://prettier.io/ for web?

2. I think simple `cargo check` & `cargo test` is enough in this case. Both of them are already triggered as Github actions with a positive result ;)

---
I think this feature would be merged to ibtida as a second milestone. My suggestion would be to have a parallel feature branch eg. `feature/nft-fractionalization`, @nblogist please confirm.

Apart of that, once @nblogist accept, feel free to merge. All looks good to me!

**Comment by justinfrevert** [comment]
@korzewski 
> Is it something like https://prettier.io/ for web?

I've used prettier, and looking at this `cargo fmt` tool it seems similar enough. 

I can create a ticket, I'm not sure if it would require any research, but playing around with the tool so far it's been easy enough to find a config that mostly follows our style guide.

**Comment by aelesbao** [comment]
@justinFrevert about `rust fmt`, I configured it with the coding guidelines [at the beginning of the project](https://github.com/anmolnetwork/anmol-node/commit/f74a3b3de44728065c86bcf80ff481d8b1b4fa2b) but never applied it to the code forked from the template. I think it would be worth fixing the code in a separate PR and run it as a linting step during the build. I'll create an issue for that.

File: pallets/anmol-nft/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+authors = ["Anmol Developers"]
```
Comment (aelesbao) [inline pallets/anmol-nft/Cargo.toml]: Please, keep the `authors` consistent with [other packages](https://github.com/anmolnetwork/anmol-node/blob/8e678e0c694b0aa258373ae1e96dc2458737a75a/node/Cargo.toml#L2)

**Review by aelesbao — DISMISSED** [review]
LGTM otherwise!

File: pallets/anmol-nft/Cargo.toml (hunk 3 above)
Comment (justinfrevert) [reply pallets/anmol-nft/Cargo.toml]: Good catch, I corrected the field.

File: pallets/anmol-nft/Cargo.toml (hunk 1 above)
Comment (nblogist) [reply pallets/anmol-nft/Cargo.toml]: >The goal of this is to bring the NFT logic to a place where we can modify it and add logic

Given that this is supposed to be the general NFT pallet from where we extend other pallets `anmol-nft` sounds good but what would add clarity is  `base-nft`

But for fractionalized NFT `fractionalized-nft` would be good and convey what each pallet is meant to be, which can be extended from `base-nft` 

(base-nft suggested by Phaedra)

File: pallets/anmol-nft/Cargo.toml (hunk 2 above)
Comment (nblogist) [reply pallets/anmol-nft/Cargo.toml]: See if description needs to be updated after this: https://github.com/anmolnetwork/anmol-node/pull/32#discussion_r657830398

**Comment by nblogist** [comment]
>I think this feature would be merged to ibtida as a second milestone. My suggestion would be to have a parallel feature branch eg. feature/nft-fractionalization, @nblogist please confirm.

Agreed! @justinFrevert please rename the base branch to match the [coding guidelines](https://www.notion.so/Coding-Guidelines-3213287620504a6cac84392f9f1892b7)  and branch it out from `ibtida` branch once #34 is merged.

File: pallets/nft/Cargo.toml [hunk 4]
```diff
@@ -21,7 +21,7 @@ frame-benchmarking = { default-features = false, optional = true, version = '3.1
 sp-std = { default-features = false, version = '3.0.0' }
 sp-runtime = { default-features = false, version = '3.0.0' }
 sp-core = { default-features = false, version = '3.0.0' }
-orml-nft = { version = "0.4.0", default-features = false }
+anmol-nft = { version = "0.1.0", default-features = false, path = '../anmol-nft' }
```
Comment (nblogist) [inline pallets/nft/Cargo.toml:24]: https://github.com/anmolnetwork/anmol-node/pull/32/files#r656636846

**Review by nblogist — COMMENTED** [review]
Please go through my comments @justinFrevert

File: pallets/anmol-nft/Cargo.toml (hunk 1 above)
Comment (justinfrevert) [reply pallets/anmol-nft/Cargo.toml]: Cool, I went and changed it to `base_nft` to better describe it.

File: pallets/anmol-nft/Cargo.toml (hunk 2 above)
Comment (justinfrevert) [reply pallets/anmol-nft/Cargo.toml]: I think so. I went for `A pallet to extend for building advanced NFT functions`

File: pallets/nft/src/tests.rs [hunk 5]
```diff
@@ -27,7 +27,7 @@ fn mint_nft_works() {
 		// TODO: DispatchError for ClassNotFound
 		// assert_noop!(
 		//     Nft::mint_nft(Origin::signed(ALICE), vec![0], pending_nft.clone()),
-		//     crate::Error::<Runtime>::NftError(orml_nft::Error::<Runtime>::ClassNotFound)
+		//     crate::Error::<Runtime>::NftError(base_nft::Error::<Runtime>::ClassNotFound)
```
Comment (justinfrevert) [inline pallets/nft/src/tests.rs]: Do we need the code in this comment later? I replaced the text just in case, but if not maybe we just delete it.

**Comment by justinfrevert** [comment]
@nblogist I updated with the more descriptive pallet name, the updated description, etc. Can you let me know if the changes look okay, as well as the commit/branch naming? 

FYI when going through the changes, renaming the pallet caused a lot of file diffs, as expected.

File: pallets/nft/src/tests.rs (hunk 5 above)
Comment (nblogist) [reply pallets/nft/src/tests.rs]: A good practice is to never leave blocks of code commented. If needed we can always revert back.
CC: @korzewski

**Comment by nblogist** [comment]
@justinFrevert base branch would be something like: `frevert/feature/nft-fractionalization`

File: pallets/nft/src/tests.rs (hunk 5 above)
Comment (justinfrevert) [reply pallets/nft/src/tests.rs]: I'll get rid of it

**Review by nblogist — APPROVED** [review]
Looks good to me

**Pull request merged**